### PR TITLE
fix: Setup recognizes existing npm CLI installs

### PR DESCRIPTION
**Pull request by hatayama**
## Summary
- Setup now recognizes an existing npm-installed uloop CLI instead of showing it as missing.
- Older CLI versions are surfaced as update candidates, so new projects can guide users to upgrade accurately.

## User Impact
- Before this change, Unity projects launched from a GUI environment could show "Not installed" even when the user's terminal already had uloop-cli 2.1.x installed.
- After this change, Setup runs CLI version detection through the user's login shell, matching the terminal environment where npm shims and node are available.

## Changes
- Resolve the user's login shell when Unity does not inherit SHELL.
- Run shell command discovery and uloop -v in the same login shell for macOS/Linux CLI detection.
- Treat version output as the installed-state signal while keeping the executable path as optional UI context.

## Verification
- go run ./cmd/uloop --project-path /Users/a12115/ghq/hatayama/unity-cli-loop-fix-npm-cli-detection compile
- go run ./cmd/uloop --project-path /Users/a12115/ghq/hatayama/unity-cli-loop-fix-npm-cli-detection run-tests --test-mode EditMode --filter-type regex --filter-value '.*(CliInstallationDetectorTests|NodeEnvironmentResolverTests).*'
- Verified in Unity dynamic code that SHELL=null still returns Installed=True, Version=2.1.1, and Path=/Users/a12115/.npm-global/bin/uloop

[Review Change Stack](https://app.coderabbit.ai/change-stack/hatayama/unity-cli-loop/pull/1131)

Summary
- Setup now recognizes an existing npm-installed uloop CLI by running the user’s login shell (macOS/Linux) and using the CLI's version output as the installed-state signal; the executable path remains optional UI context.
- Older CLI versions discovered via the shell are surfaced as update candidates so new projects can advise upgrades.
- Detection resolves the login shell even when Unity does not inherit SHELL (including a macOS dscl lookup fallback) and executes discovery and `uloop -v` within the same login shell environment.

User impact
- Fixes cases where Unity launched from a GUI incorrectly showed "Not installed" despite `uloop-cli` being available in the user’s terminal (e.g., npm global shims and terminal node).
- After this change, Setup reports Installed/Version/Path consistent with the terminal environment.

Key changes
- NodeEnvironmentResolver
  - Added macOS Directory Service (dscl) lookup to derive UserShell when $SHELL is missing/invalid.
  - New helpers: SelectUserShell and ExtractDirectoryServiceUserShell; GetUserShell now uses dscl fallback instead of unconditionally returning /bin/sh.
- CliInstallationDetector
  - Introduces login-shell detection on non-Windows platforms: builds a quoted login-shell command that emits start/end markers and runs it with stdout/stderr capture, a 5s timeout, and guarded process-kill behavior on cancellation/timeouts.
  - Parses marker-delimited stdout to extract an optional executable path and the CLI version; selection logic now prefers shell detection when the shell-reported version is present even if the path is absent.
  - Added a short version flag constant and uses `-v` for shell commands.
  - Refactored process-kill handling to tolerate races where the child has already exited.
- SetupWizardWindow
  - Added internal helpers to centralize skills-target UI visibility logic and updated UpdateSkillsStep to use them.
- Package removal handling
  - New UnityCliLoopPackageRemovalSettingsResetter resets only setup-wizard related settings when the package is removed; wired into InfrastructureEditorStartup so removal clears wizard auto-show state while preserving other preferences.
- Constants
  - Added CliConstants.SHORT_VERSION_FLAG = "-v".
  - Added UnityCliLoopConstants.PACKAGE_NAME = "io.github.hatayama.uloopmcp".

Tests
- Extensive new/updated NUnit tests:
  - CliInstallationDetectorTests: selection when shell version exists without path, short `-v` usage, parsing shell output (path+version and version-only), and guarded process-kill race test.
  - NodeEnvironmentResolverTests: dscl parsing and SelectUserShell behavior (env shell vs directory service vs deterministic POSIX fallback).
  - SetupWizardWindowTests: skills-target row/list visibility under first-time/setup conditions.
  - UnityCliLoopPackageRemovalSettingsResetterTests: package-name match, ShouldResetSetupWizardState behavior, and ResetSetupWizardState/ResetSetupWizardStateIfPackageRemoved persistence semantics.
  - General test hygiene: setup/teardown for editor settings file manipulation.

Verification
- Provided `go run`/test runs demonstrating scenario where SHELL=null returns Installed=True, Version=2.1.1, Path=/Users/a12115/.npm-global/bin/uloop (shell-based detection succeeds even when Unity's environment lacked SHELL).

Files touched (high level)
- Packages/src/Editor/Infrastructure/Utils/NodeEnvironmentResolver.cs (+ tests)
- Packages/src/Editor/Infrastructure/CLI/CliInstallationDetector.cs (+ tests)
- Packages/src/Editor/Domain/CliConstants.cs
- Packages/src/Editor/Presentation/Setup/SetupWizardWindow.cs (+ tests)
- Packages/src/Editor/Settings/UnityCliLoopPackageRemovalSettingsResetter.cs (+ tests)
- Packages/src/Editor/ToolContracts/UnityCliLoopConstants.cs
- Assets/Tests/Editor/* (multiple new/updated tests)
- InfrastructureEditorStartup wiring updated to register the package-removal resetter.

Review notes / risk
- Subprocess invocation, shell-quoting, dscl invocation, and timeout/kill logic are platform-sensitive; review escaping, deterministic behavior across shells, cancellation/timeout handling, and macOS dscl invocation safety (username validation already added).
- Ensure the short `-v` flag is appropriate across supported CLI versions and platforms.
- Tests increase coverage for edge cases but review runtime impacts in editor contexts (timeouts, blocking behavior).

[Review Change Stack](https://app.coderabbit.ai/change-stack/hatayama/unity-cli-loop/pull/1131)

[Review Change Stack](https://app.coderabbit.ai/change-stack/hatayama/unity-cli-loop/pull/1131)

### CODE REVIEW

File: Packages/src/Editor/Infrastructure/CLI/CliInstallationDetector.cs [hunk 1]
```diff
@@ -135,12 +140,128 @@ private static CliInstallationDetection DetectPackageOwnedCliInstallationBlockin
...
+                if (!exited)
+                {
+                    process.Kill();
+                    return null;
+                }
+
+                process.WaitForExit();
+                return outputBuilder.ToString().Trim();
+            }
+        }
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Editor/Infrastructure/CLI/CliInstallationDetector.cs:250]: Fixed in cb1257d6.

I added a shared `KillProcessIfRunning` helper and routed both cancellation and timeout cleanup through it, so an already-exited process only suppresses `InvalidOperationException`. I also added a focused regression test covering an exited process.

File: Packages/src/Editor/Infrastructure/CLI/CliInstallationDetector.cs [hunk 2]
```diff
@@ -135,12 +140,128 @@ private static CliInstallationDetection DetectPackageOwnedCliInstallationBlockin
...
+                process.ErrorDataReceived += (sender, e) => { };
+
+                process.BeginOutputReadLine();
+                process.BeginErrorReadLine();
+
+                using CancellationTokenRegistration registration = ct.Register(() => process.Kill());
+                bool exited = process.WaitForExit(PROCESS_TIMEOUT_MS);
+                if (!exited)
+                {
+                    process.Kill();
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Editor/Infrastructure/CLI/CliInstallationDetector.cs]: Fixed in cb1257d6.

I added a shared `KillProcessIfRunning` helper and routed both cancellation and timeout cleanup through it, so an already-exited process only suppresses `InvalidOperationException`. I also added a focused regression test covering an exited process.
